### PR TITLE
Simplify the DARP model

### PR DESCRIPTION
**Pull request by cquiroz**
I wasn't very happy about the DARP attachment model and I think it could be improved a bit with this PR

### CODE REVIEW

File: bundle/edu.gemini.model.p1.submit/src/main/scala/edu/gemini/model/p1/submit/Post.scala [hunk 1]
```diff
@@ -14,8 +14,8 @@ private[submit] object Post {
     con.setRequestProperty("Content-Type", s"""multipart/form-data; boundary="$BOUNDARY"""")
 
     for {
-      pdf1 <- p.meta.attachments.sortBy(_.index).map(_.name).headOption.flatten.toRight(ClientError("The proposal is missing some PDF attachment.")).right
-      pdf2 <- p.meta.attachments.sortBy(_.index).map(_.name).lastOption.flatten.toRight(ClientError("The proposal is missing some PDF attachment.")).right
+      pdf1 <- p.meta.firstAttachment.toRight(ClientError("The proposal is missing some PDF attachment.")).right
+      pdf2 <- p.meta.secondAttachment.toRight(ClientError("The proposal is missing some PDF attachment.")).right
```
Comment (swalker2m) [inline bundle/edu.gemini.model.p1.submit/src/main/scala/edu/gemini/model/p1/submit/Post.scala:18]: Can/should we be more specific about which PDF attachment is missing?

File: bundle/edu.gemini.model.p1/src/test/resources/edu/gemini/model/p1/immutable/proposal_itac_no_ngoauthority.xml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <proposal tacCategory="Extragalactic" schemaVersion="1.0.0">
     <meta>
-        <attachment name="ar_2012B_007.pdf" index="1"/>
-        <attachment name="ar_2012B_007.pdf" index="2"/>
```
Comment (swalker2m) [inline bundle/edu.gemini.model.p1/src/test/resources/edu/gemini/model/p1/immutable/proposal_itac_no_ngoauthority.xml:5]: they can be the same file in the end?

**Review by swalker2m — APPROVED** [review]
Seems like a great improvement.  :+1:

File: bundle/edu.gemini.model.p1.submit/src/main/scala/edu/gemini/model/p1/submit/Post.scala (hunk 1 above)
Comment (cquiroz) [reply bundle/edu.gemini.model.p1.submit/src/main/scala/edu/gemini/model/p1/submit/Post.scala:18]: Yes, good idea

File: bundle/edu.gemini.model.p1/src/test/resources/edu/gemini/model/p1/immutable/proposal_itac_no_ngoauthority.xml (hunk 2 above)
Comment (cquiroz) [reply bundle/edu.gemini.model.p1/src/test/resources/edu/gemini/model/p1/immutable/proposal_itac_no_ngoauthority.xml:5]: So this is just a test, and currently there is no restriction. But it is a good question whether we should verify it

**Pull request merged**